packages/shared/lib/core/wallet/utils/getDepositAddress.ts

```diff
@@ -1,12 +1,14 @@
 import { AccountAddress, AccountOutput } from '@iota/sdk/out/types'
 import { IWallet } from '../../profile/interfaces'
 import { getBech32AddressFromAddressTypes } from './getBech32AddressFromAddressTypes'
+import { updateMainAccountId } from '../stores'
 
 // TODO(2.0): Update the implementation to handle multiple accounts and use the selected account's address.
 export async function getDepositAddress(wallet: IWallet): Promise<string> {
     const accounts = await wallet?.accounts()
     if (accounts?.length) {
         const accountId = (accounts[0]?.output as AccountOutput).accountId
+        updateMainAccountId(accountId)
```

Comment:
the same as the first comment, I think it is not the place to update de main account id